lib/tests/helpers.test.js

```diff
@@ -216,4 +218,50 @@ describe("Submit and Template helpers", () => {
     expect(mockedFormikBag.setStatus).toBeCalledTimes(1);
     expect(mockedFormikBag.setStatus).toBeCalledWith("Error");
   });
+
+  test("Should submit with the correct csrf token", async () => {
```

Comment:
When going through the current code base I noticed Jest Tests is something that we really need to work on as a team.  I'll use this as an example but by no means is it the only one in our codebase.
The way the current test is written goes beyond the scope of what the client side function should test for.  The additional scope also doesn't add much value since all of the server side logic and responses are mocked out.
For example:
- the test mocks the axios server response as succeeded so `expect(mockedRouter.push).toBeCalledTimes(1)` will always evaluate to true.
- testing the payload `buildFormDataObject(TestForm, mockResponses)` is redundant and will always evaluate to true as it's relying on `buildFormDataObject` to check it's generated output against itself.

The test could be written to test solely that the Submit API function is correctly adding the token to the request as so:
```js
  test("Should submit with the correct csrf token", async () => {
    getCsrfToken.mockReturnValueOnce("csrfTokenSecretValue");

    mockedAxios.mockResolvedValue({
      status: 200,
      data: { received: true },
    });

    await submitToAPI(mockResponses, {
      props: {
        formConfig: TestForm,
        language: "en",
        router: { push: jest.fn() },
      },
      setStatus: jest.fn(),
    });
    expect(mockedAxios.mock.calls[0][0].headers["X-CSRF-Token"]).toBe("csrfTokenSecretValue");
  });

```